cpp/src/io/json/read_json.cu

```diff
@@ -103,18 +115,34 @@ rmm::device_uvector<char> ingest_raw_input(host_span<std::unique_ptr<datasource>
     }
 
     stream.synchronize();
-    return d_buffer;
+    // If input JSON buffer has single quotes and option to normalize single quotes is enabled,
```

Comment:
I'd rather have the preprocess step directly in `read_json`, outside of `ingest_raw_input`.

Reply:
Good idea, I've moved the check for the single quote normalization option to `read_json`